### DISCUSSION TITLE
which scheduler is selected?

### DISCUSSION THREAD

**Discussion by choppsv1** [pytest-dev/pytest-xdist#692] [General]
During setup (deep inside a call chain, far from the actual module setup function) I need to create a log/tmp directory path for the tests that will be running. I want this to work for `--dist=loadfile` as well as `--dist=each`. For the `each` case I will add the worker as a sub-directory so each run does not conflict; however, for the `loadfile` case (which most people will use) I do not want to do that, as the added "gw0", "gw1", ... directory just obfuscates things.

So, what I need to do is a way to determine which type of scheduler is being run (or some indication that multiple of each test can be running at the same time). How could I do this? I do have a `pytest_configure()` in my conftest.py if that might be one place to check?

**Comment by nicoddemus** [comment]
One solution would be to look at the option itself:

```python
import os
def pytest_configure(config):
    if "PYTEST_XDIST_WORKER" in os.environ:
        return
    if config.getoption("distmode") == "each":
        ...
    elif config.getoption("distmode") == "load":
        ...
```

Note that this will only work on the `master` node.

**Reply by choppsv1** [reply]
Right I needed the info in the worker (to determine how to create log direcotry). I used your exampel (changed "distmode" to "dist") and set an environment variable which did get passed to the worker.
```
    if "PYTEST_XDIST_WORKER" not in os.environ:
        os.environ["PYTEST_XDIST_MODE"] = config.getoption("dist", "no")
 ```

**Reply by nicoddemus** [reply]
That's a good solution (also demonstrates how to pass data between main and the worker). I think you can mark your own reply as the answer to the question then. 👍

**Reply by choppsv1** [reply]
> That's a good solution (also demonstrates how to pass data between main and the worker). I think you can mark your own reply as the answer to the question then. 👍

It' not offering me an option to mark as the answer oddly.
